File: Backend/twetter/src/main/java/com/bartdebever/twetter/services/Local/LocalUserService.java
package com.bartdebever.twetter.services.Local;

import com.bartdebever.twetter.models.User;
import com.bartdebever.twetter.services.interfaces.IUserService;

public class LocalUserService extends LocalCrudService<User> implements IUserService {
    public void addFollow(int followerId, int followingId) {

    }

    @Override
    public User searchByName(String username) {
        for(User user : getEntityList()) {
            if (user.getUserName().equals(username)) {
                return user;
            }
        }

        return null;
    }
}

File: Backend/src/main/kotlin/services/local/LocalUserService.kt
package services.local

import models.User
import services.interfaces.IUserService
import javax.enterprise.context.RequestScoped
import javax.enterprise.inject.Alternative
import javax.inject.Named


@RequestScoped
@Named("UserTest")
@Alternative
class LocalUserService : LocalCrudService<User>(), IUserService {
    override fun validateUser(username: String, password: String): User? {
        for (user in entityList) {
            if (user.userName == username && user.password == password) {
                return user
            }
        }

        return null
    }

    override fun addFollow(followerId: Int, followingId: Int) {

    }

    override fun searchByName(username: String): User? {
        for (user in entityList) {
            if (user.userName == username) {
                return user
            }
        }

        return null
    }
}
